torch/optim/lr_scheduler.py

```diff
@@ -0,0 +1,279 @@
+from bisect import bisect_right
+from .optimizer import Optimizer
+
+
+class _LRScheduler(object):
+    def __init__(self, optimizer, last_epoch=-1):
+        if not isinstance(optimizer, Optimizer):
+            raise TypeError('{} is not an Optimizer'.format(
+                type(optimizer).__name__))
+        self.optimizer = optimizer
+        if last_epoch == -1:
+            for group in optimizer.param_groups:
+                group.setdefault('initial_lr', group['lr'])
+        else:
+            for i, group in enumerate(optimizer.param_groups):
+                if 'initial_lr' not in group:
+                    raise KeyError("param 'initial_lr' is not specified "
+                                   "in param_groups[{}] when resuming an optimizer".format(i))
+        self.base_lrs = list(map(lambda group: group['initial_lr'], optimizer.param_groups))
+        self.step(last_epoch + 1)
+        self.last_epoch = last_epoch
+
+    def get_lr(self):
+        raise NotImplementedError
+
+    def step(self, epoch=None):
+        if epoch is None:
+            epoch = self.last_epoch + 1
+        self.last_epoch = epoch
+        for param_group, lr in zip(self.optimizer.param_groups, self.get_lr()):
+            param_group['lr'] = lr
+
+
+class LambdaLR(_LRScheduler):
+    def __init__(self, optimizer, lr_lambda, last_epoch=-1):
+        self.optimizer = optimizer
+        self.base_lrs = list(map(lambda group: group['lr'], optimizer.param_groups))
```

Comment:
Why is it reading `lr` here? What's wrong with letting `_LRScheduler` figure out what should be the base LRs? All attributes assigned in this constructor (except `lr_lambdas`) will be overwritten by the super constructor, so there's no need to do it.